src/jit/compiler.h

```diff
@@ -7992,7 +7993,10 @@ public :
         // Methods returning a struct in two registers is considered having a return value of TYP_STRUCT.
         // Such method's compRetNativeType is TYP_STRUCT without a hidden RetBufArg
         return varTypeIsStruct(info.compRetNativeType) && (info.compRetBuffArg == BAD_VAR_NUM);
-#else 
+#elif FEATURE_MULTIREG_RET && defined(_TARGET_X86_)
```

Comment:
For readability, I would suggest to organize this as follows

#if FEATURE_MULTIREG_RET

#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING) || defined(_TARGET_ARM)
    return varTypeIsStruct(info.compRetNativeType) && (info.compRetBuffArg == BAD_VAR_NUM);
#else if defined (_TARGET_X86_)
    return varTypeIsLong(info.compRetNativeType);
#else 
    unreached();
#endif

#else 
   return false;
#endif